### ISSUE TITLE
Saving a --server based connection in interactive mode generates incorrect connections file

### ISSUE THREAD

**Issue by andy-maier** [pywbem/pywbemtools#396]
Example (the selection of a current connection at the begin is only needed to circumvent issue #395):
```
$ pywbemcli
pywbemcli> connection select mymock
"mymock" current
pywbemcli> connection list
WBEM server connections: (#: default, *: current)
+-----------+------------------+-------------+--------+-----------+----------+----------------------------------------+
| name      | server           | namespace   | user   |   timeout | verify   | mock-server                            |
|-----------+------------------+-------------+--------+-----------+----------+----------------------------------------|
| *mymock   |                  | root/cimv2  |        |        30 | True     | tests/unit/simple_assoc_mock_model.mof |
| bla       | http://blahblah  | root/cimv2  |        |        30 | True     |                                        |
| local     | http://localhost | root/cimv2  | me     |        30 | True     |                                        |
| mockfuzzy |                  | root/cimv2  |        |        30 | True     | tests/unit/simple_mock_model.mof       |
|           |                  |             |        |           |          | tests/unit/simple_mock_invokemethod.py |
+-----------+------------------+-------------+--------+-----------+----------+----------------------------------------+
pywbemcli> --server http://localhost --user usr1 --password blah connection save testconn
pywbemcli> connection list
Traceback (most recent call last):
  File "/Users/maiera/virtualenvs/pywbem27/bin/pywbemcli", line 10, in <module>
    sys.exit(cli())
  File "/Users/maiera/virtualenvs/pywbem27/lib/python2.7/site-packages/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/Users/maiera/virtualenvs/pywbem27/lib/python2.7/site-packages/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/Users/maiera/virtualenvs/pywbem27/lib/python2.7/site-packages/click/core.py", line 1114, in invoke
    return Command.invoke(self, ctx)
  File "/Users/maiera/virtualenvs/pywbem27/lib/python2.7/site-packages/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/maiera/virtualenvs/pywbem27/lib/python2.7/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/Users/maiera/virtualenvs/pywbem27/lib/python2.7/site-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/Users/maiera/PycharmProjects/pywbem/pywbemtools/pywbemtools/pywbemcli/pywbemcli.py", line 550, in cli
    ctx.invoke(repl)
  File "/Users/maiera/virtualenvs/pywbem27/lib/python2.7/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/Users/maiera/virtualenvs/pywbem27/lib/python2.7/site-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/Users/maiera/PycharmProjects/pywbem/pywbemtools/pywbemtools/pywbemcli/pywbemcli.py", line 601, in repl
    click_repl.repl(ctx, prompt_kwargs=prompt_kwargs)
  File "/Users/maiera/virtualenvs/pywbem27/lib/python2.7/site-packages/click_repl/__init__.py", line 248, in repl
    group.invoke(ctx)
  File "/Users/maiera/virtualenvs/pywbem27/lib/python2.7/site-packages/click/core.py", line 1137, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/maiera/virtualenvs/pywbem27/lib/python2.7/site-packages/click/core.py", line 1137, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/maiera/virtualenvs/pywbem27/lib/python2.7/site-packages/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/maiera/virtualenvs/pywbem27/lib/python2.7/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/Users/maiera/virtualenvs/pywbem27/lib/python2.7/site-packages/click/decorators.py", line 27, in new_func
    return f(get_current_context().obj, *args, **kwargs)
  File "/Users/maiera/PycharmProjects/pywbem/pywbemtools/pywbemtools/pywbemcli/_cmd_connection.py", line 233, in connection_list
    context.execute_cmd(lambda: cmd_connection_list(context))
  File "/Users/maiera/PycharmProjects/pywbem/pywbemtools/pywbemtools/pywbemcli/_context_obj.py", line 206, in execute_cmd
    cmd()
  File "/Users/maiera/PycharmProjects/pywbem/pywbemtools/pywbemtools/pywbemcli/_cmd_connection.py", line 233, in <lambda>
    context.execute_cmd(lambda: cmd_connection_list(context))
  File "/Users/maiera/PycharmProjects/pywbem/pywbemtools/pywbemtools/pywbemcli/_cmd_connection.py", line 542, in cmd_connection_list
    svr.keyfile, "\n".join(svr.mock_server)]
TypeError: can only join an iterable
```

Investigating the connection definitions file reveals that the connection testconn that was successfully saved, has `mock-server: null` instead of `mock-server: []`, and that causes the exception shown above. Fixing that manually in the connections definition file causes the exception to go away:
```
    testconn:
        default-namespace: root/cimv2
        user: usr1
        password: blah
        ca-certs:
        - /etc/pki/ca-trust/extracted/openssl/ca-bundle.trust.crt
        - /etc/ssl/certs
        - /etc/ssl/certificates
        mock-server: null           <-- should be [] instead of null
        certfile: null
        name: testconn
        verify: true
        server: http://localhost
        timeout: 30
        keyfile: null
```

Note that saving a new --server based connection **in command mode** sets `mock-server` correctly to `[]`.

**Comment by KSchopmeyer** [comment]
Bad code.  Think this is a partner to #395

**Comment by andy-maier** [comment]
It appears this issue has also been solved by PR #403.
